xlml/apis/task.py

```diff
@@ -374,6 +377,25 @@ def run_model(self) -> DAGNode:
     """
     with TaskGroup(group_id="run_model") as group:
       workload_id = xpk.generate_workload_id(self.task_test_config.benchmark_id)
+      gcs_path = name_format.generate_gcs_folder_location(
+          self.task_test_config.benchmark_id
+      )
+      launch_workload = self.launch_workload(workload_id, gcs_path)
+      wait_for_workload_completion = xpk.wait_for_workload_completion.override(
+          timeout=self.task_test_config.time_out_in_min * 60,
+      )(
+          workload_id=workload_id,
+          project_id=self.task_gcp_config.project_name,
+          region=self.task_gcp_config.zone[:-2],
+          cluster_name=self.task_test_config.cluster_name,
+      )
+
+      (workload_id, gcs_path) >> launch_workload >> wait_for_workload_completion
```

Comment:
You don't need to add an explicit control dependency for `workload_id` and `gcs_path` because `launch_workload` already consumes the results.

Reply:
I discussed with Ran, since we already have the dependency chain we wanted to keep this part explicit as well. I don't mind either way.

Reply:
Yeah, either remove this line or keep it correct for less confusion.